lib/Sema/PreCheckExpr.cpp

```diff
@@ -519,7 +519,15 @@ Expr *TypeChecker::resolveDeclRefExpr(UnresolvedDeclRefExpr *UDRE,
         diag.highlight(UDRE->getSourceRange());
         typo->addFixits(diag);
       } else {
-        emitBasicError();
+        if (Name.isSimpleName("async")) {
+          auto diag = Context.Diags.diagnose(Loc,
+              diag::cannot_find_in_scope_corrected, Name,
+              /*isOperator=*/false, "await");
+          diag.highlight(UDRE->getSourceRange());
+          diag.fixItReplace(Loc, "await");
```

Comment:
I think you should be able to chain the method calls (in case you'd like to avoid creating a binding for `diag`).

Reply:
This is what's done immediately above. If you want it chained, I'm not terribly partial so I can make that change, but I do want to stay consistent.

Reply:
Ah okay. Yeah, it was just FYI (in case you missed it). Creating a binding is fine by me. 👍🏼